github_api_helper.js

```diff
@@ -27,24 +28,41 @@ function inProgressAPIresponse (context, headSha) {
 }
 
 /**
- * Filters the listed files and returns only the files relevant to us
- * @param {*} rawData the raw output from list files API call
- * @return {Promise[]<any>} the filtered GitHub response
+ * Get Precaution config file contents as raw data,
+ * if that file exists.
+ * @param {import('probot').Context} context Probot context
+ * @return {Promise<any>} GitHub response
+ * See https://developer.github.com/v3/repos/contents/#get-contents
  */
-function filterData (rawData) {
-  return rawData.data.filter(file => config.fileExtensions.reduce((acc, ext) => acc || file.filename.endsWith(ext), false))
-    .filter(fileJSON => fileJSON.status !== 'removed').map(fileInfo => {
-      return fileInfo.filename
+async function getConfigFile (context) {
+  let { owner, repo } = context.repo()
+  // GitHub doesn't provide "checkIfFileExists" API endpoint.
+  // That's why we should try to download the file and if we have
+  // 404 error code the file doesn't exist in the repository.
+  try {
+    return await context.github.repos.getContents({
+      owner,
+      repo,
+      path: config.configFilePath,
+      headers: { accept: rawMediaType }
     })
+  } catch (error) {
+    if (error.code === 404) {
+      return undefined
```

Comment:
Based on the other discussion around `undefined` vs `null` we should probably be returning `null` here